### DISCUSSION TITLE
Looking for Air repackager for older air files and possible AIr phone.

### DISCUSSION THREAD

**Discussion by Gametrek** [airsdk/Adobe-Runtime-Support#2663] [Ideas]
Air seems to prevent older .air from running wiithout being repackaged. I am looking for a method of repackaging using 

A. Flash
B. The SDK I have installed
C. A method that is reachable without updating anything else. 

Another things I have been reading about is iOS and the fact that it is really one big HTML/JAva system. So why not an AIR phone 
We need to break away from standards and have unique creations. Android is just Linux. 

That being said an Air application that is able to read an SIM card and connect to the phone service ( with internet of course ) on your desktop ( etc ) and or be a device no different then any other phone. It would be so unique to break away and have a fully functioning flash based phone. Just an idea.

**Comment by ajwfrost** [comment]
> Air seems to prevent older .air from running wiithout being repackaged

It shouldn't! The only issue you might run into is if the older .air app needed the HTML/JS capabilities, in which case you'd need to use the 33.1.1.744 release of the AIR runtime, as we took out the old WebKit library after that. Plus it's possible that in the past, someone created an air app that used html and also Flash within this, and of course this wouldn't work any more as (almost) no one has the appropriate Flash Player installed on their computers now.

An AIR phone would be interesting :-) we worked on a "Flash Home" project for Adobe many years ago, where the home screen and main UI elements (call logs, messages, settings etc) were all done using Flash Lite. But I suspect it would take a lot to try to break into this sort of thing; I would also say that without some redesign of the animation engine, Flash/AIR is not so energy efficient if it's meant to sit there not doing anything for a while.. 

thanks
